### PR TITLE
DOC: fix DataFrame.nunique docstring and doctests

### PR DESCRIPTION
**Pull request by Moisan**
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Based on #22459. Fix the docstring for DataFrame.nunique. I also updated `ci/code_checks.sh`.

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -7279,32 +7279,42 @@ def f(x):
 
     def nunique(self, axis=0, dropna=True):
         """
-        Return Series with number of distinct observations over requested
-        axis.
+        Return Series with number of distinct observations.
+
+        Count distinct observations over requested axis. Can ignore NaN values.
```
Comment (gfyoung) [inline pandas/core/frame.py]: Sorry about this, but this sentence would work better as the summary i.e.

~~~
Count distinct observations over requested axis.

Return Series with number of distinct observations on the provided axis.  Can ignore NaN values.
~~~

The reason for my preference is that the summary sentence as I have swapped it is now a little less jargon-y (i.e. no mention of pandas-specific objects).  This is *slightly* nicer to read for an end-user if they just want the short version of what this method does.

Comment (Moisan) [reply pandas/core/frame.py]: No problem :).

**Review by gfyoung — APPROVED** [review]
cc @datapythonista

**Comment by datapythonista** [comment]
Thanks @Moisan

**Pull request merged**